src/main/scala/org/hammerlab/magic/test/listener/TestSparkListener.scala

```diff
@@ -158,7 +158,12 @@ object TestSparkListener {
 
   var instance: TestSparkListener = _
 
-  def apply(): TestSparkListener = instance
+  def apply(): TestSparkListener = {
+    if (instance == null) {
+      instance = new TestSparkListener()
```

Comment:
I'm worried about this change, because the only code instantiating a `TestSparkListener` should be inside SparkContext, where it creates one based on the `spark.extraListeners` conf; making our own here seems like a recipe for being out of sync with the one our SparkContext actually uses.